lib/DocumentsFieldArray/DocumentsFieldArray.js

```diff
@@ -117,79 +57,14 @@ class DocumentsFieldArray extends React.Component {
         key={i}
         onDelete={() => onDeleteField(i, doc)}
       >
-        <Row>
-          <Col xs={12} md={onUploadFile ? 6 : 12}>
-            <Row>
-              <Col xs={12}>
-                <Field
-                  data-test-document-field-name
-                  component={TextField}
-                  id={`${name}-name-${i}`}
-                  label={<FormattedMessage id="stripes-erm-components.doc.name" />}
-                  name={`${name}[${i}].name`}
-                  required
-                  validate={this.validateRequired}
-                />
-              </Col>
-            </Row>
-            {this.renderCategory(i)}
-            <Row>
-              <Col xs={12}>
-                <Field
-                  data-test-document-field-note
-                  component={TextArea}
-                  id={`${name}-note-${i}`}
-                  label={<FormattedMessage id="stripes-erm-components.doc.note" />}
-                  name={`${name}[${i}].note`}
-                  parse={v => v} // Lets us send an empty string instead of `undefined`
-                />
-              </Col>
-            </Row>
-          </Col>
-          {onUploadFile &&
-            <Col xs={12} md={6}>
-              <Field
-                component={FileUploaderField}
-                data-test-document-field-file
-                id={`${name}-file-${i}`}
-                label={<FormattedMessage id="stripes-erm-components.doc.file" />}
-                name={`${name}[${i}].fileUpload`}
-                onDownloadFile={onDownloadFile}
-                onUploadFile={onUploadFile}
-                validate={this.validateDocIsSpecified}
-              />
-            </Col>
-          }
-        </Row>
-        <Row>
-          <Col xs={12}>
-            <Field
-              data-test-document-field-location
-              component={TextField}
-              id={`${name}-location-${i}`}
-              label={<FormattedMessage id="stripes-erm-components.doc.location" />}
-              name={`${name}[${i}].location`}
-              parse={v => v} // Lets us send an empty string instead of `undefined`
-              validate={this.validateDocIsSpecified}
-            />
-          </Col>
-        </Row>
-        <Row>
-          <Col xs={12}>
-            <Field
-              component={TextField}
-              data-test-document-field-url
-              id={`${name}-url-${i}`}
-              label={<FormattedMessage id="stripes-erm-components.doc.url" />}
-              name={`${name}[${i}].url`}
-              parse={v => v} // Lets us send an empty string instead of `undefined`
-              validate={composeValidators(
-                this.validateDocIsSpecified,
-                this.validateURLIsValid,
-              )}
-            />
-          </Col>
-        </Row>
+        <Field
+          component={DocumentField}
+          documentCategories={documentCategories}
+          index={i}
+          name={`${name}`}
```

Comment:
OK wait a sec here, shouldn't this be something like the following?
```suggestion
          name={`${name}[${i}]`}
```

That'll allow you to avoid [this weird indexing](https://github.com/folio-org/stripes-erm-components/pull/168/files#diff-556c68713cd130583f07663def58a56fR40).

Reply:
Yeah you are right. Although, this would need a lot of changes in the tests I believe, such as changing all the selectors names https://github.com/folio-org/ui-agreements/blob/master/test/ui-testing/docs.js#L52. So its either this vs that?